app/models/mixins/has_monitoring_manager_mixin.rb

```diff
@@ -5,17 +5,15 @@ module HasMonitoringManagerMixin
 
   def ensure_monitoring_manager
     # monitoring_manager should be defined by child classes.
-    if try(:monitoring_manager_needed?)
+    if try(:monitoring_manager_needed?) && monitoring_manager.nil?
       build_monitoring_manager(:parent_manager => self)
       monitoring_manager.name = "#{name} Monitoring Manager"
-    end
-    ensure_monitoring_manager_properties
-  end
-
-  def ensure_monitoring_manager_properties
-    if monitoring_manager
       monitoring_manager.zone_id = zone_id
       monitoring_manager.provider_region = provider_region
     end
   end
+
+  def ensure_monitoring_manager_deletion
+    monitoring_manager.try(:delete_queue) unless try(:monitoring_manager_needed?)
```

Comment:
if `try(:monitoring_manager_needed?)` fails — if it doesn't respond to such method — is it correct to delete?

Reply:
I think the openshift provider is the only one who uses the monitoring manager so its up to us. As best as i can tell, yes. Because if you dont have the endpoint\role for monitoring then you dont need the manager.
We wont lose data the collected since the `MiqAlerts` are associated with the parent ems and not this manager.

Reply:
yes these should only be called for container managers, and they all implement `monitoring_manager_needed?` since it's in container manager mixin. I suspect this was only done to allow merging the manageiq PR first with out breaking anything. @zeari can we loose the try?